google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/models/TableAdminRequests.java

```diff
@@ -0,0 +1,236 @@
+import com.google.protobuf.ByteString;
+
+/** Fluent DSL models to build Bigtable Table admin requests */
+@BetaApi
+public final class TableAdminRequests {
+  private TableAdminRequests() {}
+
+  /**
+   * Factory method to get an instance of CreateTable
+   *
+   * @param tableId - the id of the table to create
+   * @return CreateTable
+   */
+  public static CreateTable createTable(String tableId) {
+    return new CreateTable(tableId);
+  }
+
+  /**
+   * Factory method to get an instance of ModifyFamilies
+   *
+   * @param tableId - the id of the table to create
+   * @return ModifyFamilies
+   */
+  public static ModifyFamilies modifyFamilies(String tableId) {
+    return new ModifyFamilies(tableId);
+  }
+
+  /**
+   * Fluent wrapper for {@link CreateTableRequest}
+   *
+   * <pre>
+   * Allows for creating table with
+   *    - optional columnFamilies, including optional {@link GCRule}
+   *    - optional granularity
+   *    - and optional split points
+   * </pre>
+   */
+  public static final class CreateTable {
+    private final CreateTableRequest.Builder createTableRequest = CreateTableRequest.newBuilder();
+    private final Table.Builder tableRequest = Table.newBuilder();
+
+    /**
+     * Configures table with the specified id
+     *
+     * @param tableId
+     */
+    private CreateTable(String tableId) {
+      createTableRequest.setTableId(tableId);
+    }
+
+    /**
+     * Configures table with the specified granularity
+     *
+     * @param granularity
+     */
+    public CreateTable withGranularity(TimestampGranularity granularity) {
+      Preconditions.checkNotNull(granularity);
+      tableRequest.setGranularity(granularity);
+      return this;
+    }
+
+    /**
+     * Adds a new columnFamily to the configuration
+     *
+     * @param familyId
+     */
+    public CreateTable addColumnFamily(String familyId) {
+      Preconditions.checkNotNull(familyId);
+      tableRequest.putColumnFamilies(familyId, ColumnFamily.newBuilder().build());
+      return this;
+    }
+
+    /**
+     * Adds a new columnFamily with {@link GCRule} to the configuration
+     *
+     * @param familyId
+     * @param gcRule
+     */
+    public CreateTable addColumnFamily(String familyId, GCRule gcRule) {
+      Preconditions.checkNotNull(familyId);
+      tableRequest.putColumnFamilies(
+          familyId, ColumnFamily.newBuilder().setGcRule(gcRule.toProto()).build());
+      return this;
+    }
+
+    /**
+     * Adds split at the specified key to the configuration
+     *
+     * @param key
+     */
+    public CreateTable addSplit(ByteString key) {
+      Preconditions.checkNotNull(key);
+      createTableRequest.addInitialSplits(Split.newBuilder().setKey(key).build());
+      return this;
+    }
+
+    /**
+     * Adds the specified number of uniform splits to the configuration
+     *
+     * @param key
+     * @param numSplits
+     */
+    public CreateTable addUniformSplits(ByteString key, int numSplits) {
```

Comment:
What does key do here? I think that you probably want this be a `set` with a range and numSplits. But lets leave this off until there is an impl